OverlayDDU.h

```diff
@@ -257,26 +258,71 @@ class OverlayDDU : public Overlay
                 const float rpmPct = (rpm-lo) / (hi-lo);
 
                 const float ww = 0.16f;
-                for( int i=0; i<8; ++i )
+                if (!(ir_EngineWarnings.getInt() & irsdk_pitSpeedLimiter))
                 {
-                    const float lightPct = i/8.0f;
-                    const float lightRpm = lo + (hi-lo) * lightPct;
+                    for (int i = 0; i < 8; ++i)
+                    {
+                        const float lightPct = i / 8.0f;
+                        const float lightRpm = lo + (hi - lo) * lightPct;
 
-                    D2D1_ELLIPSE e = { float2(r2ax(0.5f-ww/2+(i+0.5f)*ww/8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                        D2D1_ELLIPSE e = { float2(r2ax(0.5f - ww / 2 + (i + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
 
-                    if( rpmPct < lightPct ) {
-                        m_brush->SetColor( outlineCol );
-                        m_renderTarget->DrawEllipse( &e, m_brush.Get() );
+                        if (rpmPct < lightPct) {
+                            m_brush->SetColor(outlineCol);
+                            m_renderTarget->DrawEllipse(&e, m_brush.Get());
+                        }
+                        else {
+                            if (lightRpm < ir_session.rpmSLFirst)
+                                m_brush->SetColor(float4(1, 1, 1, 1));
+                            else if (lightRpm < ir_session.rpmSLLast)
+                                m_brush->SetColor(warnCol);
+                            else
+                                m_brush->SetColor(float4(1, 0, 0, 1));
+                            m_renderTarget->FillEllipse(&e, m_brush.Get());
+                        }
+                    }
+                }
+                else
+                {
+                    D2D1_ELLIPSE l0 = { float2(r2ax(0.5f - ww / 2 + (0 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l1 = { float2(r2ax(0.5f - ww / 2 + (1 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l2 = { float2(r2ax(0.5f - ww / 2 + (2 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l3 = { float2(r2ax(0.5f - ww / 2 + (3 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l4 = { float2(r2ax(0.5f - ww / 2 + (4 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l5 = { float2(r2ax(0.5f - ww / 2 + (5 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l6 = { float2(r2ax(0.5f - ww / 2 + (6 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                    D2D1_ELLIPSE l7 = { float2(r2ax(0.5f - ww / 2 + (7 + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+
+                    int frames = 60;
```

Comment:
Agreed, I will take a look

Reply:
I might be misremembering about blinking used elsewhere.  It was in there at some point for something, but I might have removed it.

Reply:
For now I have opted for using the current tick count to work out if the lights should be on or off. Only thing is that I had to use a magic number (16) because the tick count was jumping 16 for every render call, however on odd occasions it would jump 17. There's definitely better ways of doing it, but this illuminates the need for incrementing a variable every tick.

The main rpm lights are moved out of the pit limiter if statement scope now, so this shouldn't affect performance at all if not using the limiter. I have fully tested this now on a laptop with lowest possible graphics settings, if anything would have performance issues it is this laptop. Tested on both 30Hz and 60Hz. I believe this is ready for merge.